libnmstate/prettystate.py

```diff
@@ -112,30 +95,45 @@ def represent_unicode(_, data):
     """
 
     return yaml.ScalarNode(
-        tag=u"tag:yaml.org,2002:str", value=data.encode("utf-8")
+        tag="tag:yaml.org,2002:str", value=data.encode("utf-8")
     )
 
 
-def order_iface_state(iface_state):
-    ordered_state = OrderedDict()
-
-    for setting in ("name", "type", "state"):
-        try:
-            ordered_state[setting] = iface_state.pop(setting)
-        except KeyError:
-            pass
-
-    for key, value in order_dict(iface_state).items():
-        ordered_state[key] = value
-
-    return ordered_state
-
-
-def order_dict(dict_):
-    ordered_dict = OrderedDict()
-    for key, value in sorted(dict_.items()):
-        if isinstance(value, dict):
-            value = order_dict(value)
-        ordered_dict[key] = value
-
-    return ordered_dict
+def _sort_with_priority(data):
+    if isinstance(data, Sequence) and not isinstance(data, str):
+        return [_sort_with_priority(item) for item in data]
+    elif isinstance(data, Mapping):
+        return _sort_dict_with_priority(data)
+    else:
+        return deepcopy(data)
+
+
+def _sort_dict_with_priority(data):
+    new_data = {}
+    normal_items = {}
+    dict_items = {}
+    list_items = {}
+    for key, value in data.items():
+        if isinstance(value, Sequence) and not isinstance(value, str):
```

Comment:
Do you also need to check that it is not a unicode instance?

Reply:
I don't know when that will happen.
But even it happens, copy every bytes does not hurt.